cookbook/core/containerization/multi_images.py

```diff
@@ -1,153 +1,92 @@
-
-        docker build . -f ./<dockerfile-folder>/<dockerfile-name> -t <your-name>/<docker-image-name>:<version>
-3. Once the Docker image is built, login to your Dockerhub account from the CLI:
-
-    .. code-block::
-
-        docker login
-4. It prompts you to enter the username and the password.
-5. Push the Docker image to Dockerhub:
-
-.. code-block::
-
-   docker push <your-dockerhub-name>/<docker-image-name>
-
-Example: Suppose your Dockerfile is named `Dockerfile.prediction`, Docker image name is `multi-images-prediction` with the `latest` version, your build and push commands would look like:
-
-.. code-block::
-
-    docker build -f ./path-to-dockerfile/Dockerfile.prediction -t username/multi-images-prediction:latest
-    docker login
-    docker push dockerhub_name/multi-images-prediction
+.. note::
+    If the Docker image is not available publicly, refer to :ref:`Pulling Private Images<private_images>`.
+"""
 
-.. tip::
+import numpy as np
+from flytekit import task, workflow
 
-    Sometimes, ``docker login`` may not be successful. In such cases, execute ``docker logout`` and ``docker login``.
 
-Let's dive into the example.
-"""
-# %%
-# Import the necessary dependencies.
-from typing import NamedTuple
+@task(container_image="{{.image.mindmeld.fqn}}:{{.image.mindmeld.version}}")
+def get_data() -> np.ndarray:
+    # here we're importing scikit learn within the Flyte task
+    from sklearn import datasets
 
-import pandas as pd
-from flytekit import task, workflow
-from sklearn.model_selection import train_test_split
-from sklearn.svm import SVC
+    iris = datasets.load_iris()
+    X = iris.data[:, :2]
+    return X
 
-split_data = NamedTuple(
-    "split_data",
-    train_features=pd.DataFrame,
-    test_features=pd.DataFrame,
-    train_labels=pd.DataFrame,
-    test_labels=pd.DataFrame,
-)
 
-dataset_url = "https://raw.githubusercontent.com/harika-bonthu/SupportVectorClassifier/main/datasets_229906_491820_Fish.csv"
+@task(container_image="{{.image.borebuster.fqn}}:{{.image.borebuster.version}}")
+def normalize(X: np.ndarray) -> np.ndarray:
+    return (X - X.mean(axis=0)) / X.std(axis=0)
 
-# %%
-# Define a task that fetches data and splits the data into train and test sets.
-@task(container_image="{{.image.trainer.fqn }}:{{.image.trainer.version}}")
-def svm_trainer() -> split_data:
-    fish_data = pd.read_csv(dataset_url)
-    X = fish_data.drop(["Species"], axis="columns")
-    y = fish_data.Species
-    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.31)
-    y_train = y_train.to_frame()
-    y_test = y_test.to_frame()
 
-    return split_data(
-        train_features=X_train,
-        test_features=X_test,
-        train_labels=y_train,
-        test_labels=y_test,
-    )
+@workflow
+def multi_images_wf() -> np.ndarray:
+    X = get_data()
+    X = normalize(X=X)
+    return X
 
 
 # %%
-# .. note ::
+# Observe how the ``sklearn`` library is imported in the context of a Flyte task.
```

Comment:
can we add a little explanation on when this pattern is useful? i.e. it's useful when creating tasks in a single module where some tasks have dependencies that others don't (in this case sklearn)